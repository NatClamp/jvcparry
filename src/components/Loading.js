import React from 'react';
import { Div, Icon, Container } from 'atomize';

const Loading = () => (
  <Container data-testid="loading">
    <Div
      bg="transparent"
      d="flex"
      align="center"
      justify="center"
      minH="70vh"
    >
      <Icon name="Loading3" size="4rem" color="gray600" />
    </Div>
  </Container>
);

export default Loading;
