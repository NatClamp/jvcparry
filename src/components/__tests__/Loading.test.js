/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Loading from '../Loading';

describe('Loading component', () => {
  afterEach(cleanup);
  const history = createMemoryHistory();
  it('should render the Loading component', () => {
    render(
      <Router history={history}>
        <Loading />
      </Router>,
    );
    expect(
      screen.getByTestId('loading'),
    ).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <Loading />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
