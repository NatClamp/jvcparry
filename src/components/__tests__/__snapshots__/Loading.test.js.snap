// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should match the snapshot 1`] = `
<div
  className=""
  data-testid="loading"
>
  <div
    className=""
  >
    <svg
      className=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      viewBox="0 0 24 24"
    >
      <circle
        cx="5"
        cy="12"
        fill="#B0B7C3"
        r="2"
      >
        <animate
          attributeName="opacity"
          begin="0.1"
          dur="1s"
          repeatCount="indefinite"
          values="0;1;0"
        />
      </circle>
      <circle
        cx="12"
        cy="12"
        fill="#B0B7C3"
        r="2"
      >
        <animate
          attributeName="opacity"
          begin="0.2"
          dur="1s"
          repeatCount="indefinite"
          values="0;1;0"
        />
      </circle>
      <circle
        cx="19"
        cy="12"
        fill="#B0B7C3"
        r="2"
      >
        <animate
          attributeName="opacity"
          begin="0.3"
          dur="1s"
          repeatCount="indefinite"
          values="0;1;0"
        />
      </circle>
    </svg>
  </div>
</div>
`;
